proto/kuksabase.proto

```diff
@@ -0,0 +1,128 @@
+// Copyright Robert Bosch GmbH, 2022. Part of the Eclipse Kuksa Project.
+//
+// All rights reserved. This configuration file is provided to you under the
+// terms and conditions of the Eclipse Distribution License v1.0 which
+// accompanies this distribution, and is available at
+// http://www.eclipse.org/org/documents/edl-v10.php
+
+//This is a base proto file for databroker and kuksa-val-basic
+//function set
+
+
+syntax = "proto3";
+ 
+//I added V1 as in databroker. Is this good practice?
+package kuksa.val.v1;
+import "google/protobuf/timestamp.proto";
+
+import "kuksatypes.proto";
+
+
+
+option go_package = "/kuksa_grpc_proto";
+
+/*
+Not on autorization: We assume to send auth-token or auth-uuid in metadata
+Token is a JWT like the examples here https://github.com/eclipse/kuksa.val/tree/master/kuksa_certificates/jwt
+See also https://github.com/eclipse/kuksa.val/blob/master/doc/jwt.md
+Upon reception of auth-token, server shall generate an auth_id in metadata
+that the client can use instead of auth_token in subsequent calls
+*/
+
+// The connecting service definition.
+service kuksa_val {
+  // Returns a list of requested data points.
+  rpc get (GetRequest) returns (GetResponse) {}
+  rpc set (SetRequest) returns (SetResponse) {}
+  // Subscribe to a set of data points or conditional expressions
+    // using the Data Broker Query Syntax (described in QUERY.md)
+    // Do we want to have two calls? One simple one Query?
+    // Or OneOf in Request?
+    //
+    // Returns a stream of replies.
+    //
+    // InvalidArgument is returned if the request is malformed.
+    rpc Subscribe(SubscribeRequest) returns (stream SubscribeResponse);}
+
+//Request a number of VSS paths. RequestType needs to support at least
+//CURRENT_VALUE, no type assumes type=CURRENT_VALUE
+message GetRequest {
+    RequestType type = 1;
+    repeated string path = 2;
+  }
+
+//Status is overall status, 200 if everthing is good, if only
+//some datapoints could be delivered we shall agree on another code
+//202? 206? Suggestions? 210?
+message GetResponse {
+    repeated Value datapoints = 1;
+    Status status = 2;
+}
+
+//Status is overall status, 200 if everthing is good, if only
+//some datapoints could be delivered we shall agree on another code
+//202? 206? Suggestions? 210?
+message SetRequest {
+  RequestType type = 1;
+  repeated Value values = 2;
+}
+
+//Sets shall have a global status only
+message SetResponse {
+  Status status = 1;
```

Comment:
Could we have an optional array of values to return the failures of the individual set requests?
If all the requests went through, we can use only the _global_ status. If not, then the ones which failed can be returned in the array of values along with their corresponding failures.